src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -140,6 +154,15 @@ private static function compilePattern(Route $route, $pattern, $isHost)
                     // directly adjacent, e.g. '/{x}{y}'.
                     $regexp .= '+';
                 }
+            } else {
+                if (!preg_match('//u', $regexp)) {
+                    $useUnicode = false;
+                } elseif (!$needsUnicode && preg_match('/[\x80-\xFF]|(?<!\\\\)\\\\(?:\\\\\\\\)*+(?-i:X|[pP][\{CLMNPSZ]|x\{)/', $regexp)) {
```

Comment:
This pattern should match both `\x{` and just `\x`. For character codes < 256, \xB3 and \x{B3} are equivalent.